internal/proto/scan.go

```diff
@@ -105,3 +106,63 @@ func Scan(b []byte, v interface{}) error {
 			"redis: can't unmarshal %T (consider implementing BinaryUnmarshaler)", v)
 	}
 }
+
+
+// Scan a string slice into a custom container
+// Example:
+// var container []YourStruct; ScanSlice([]string{""},&container)
+// var container []*YourStruct; ScanSlice([]string{""},&container)
+func ScanSlice(sSlice []string, container interface{}) error {
+	val := reflect.ValueOf(container)
+	ind := reflect.Indirect(val)
+	elemType := ind.Type().Elem()
+
+	// Check the if the container is pointer to slice
+	errTyp := true
+	isElemPtr := true
+	if val.Kind() == reflect.Ptr {
+		if ind.Kind() == reflect.Slice {
+			errTyp = false
+			isElemPtr = elemType.Kind() == reflect.Ptr
+		}
+	}
+	if errTyp {
+		return fmt.Errorf("wrong object type `%s` for rows scan, need *[]*Type or *[]Type", val.Type())
```

Comment:
Go convention is to use `redis: ScanSlice(non-pointer %s)`. Similar error message is returned by json package.